py/simulator.py

```diff
@@ -0,0 +1,355 @@
+#!/usr/bin/env python3
+import sys
+import numpy as np
+import pandas as pd
+import binascii
+import socket
+import argparse
+from dataclasses import dataclass
+
+from threading import Thread
+from time import sleep
+import struct
+from ruamel.yaml import YAML
+
+parser = argparse.ArgumentParser(description='Yamcs Simulator')
+
+parser.add_argument('--file', type=argparse.FileType('r'), help='Path to a YAML file')
+parser.add_argument('--mod_file', default=None)
+parser.add_argument('--test_tm')
+
+# telemetry
+parser.add_argument('--tm_host',    type=str, default='127.0.0.1', help='TM host')
```

Comment:
Please rename all options to all hyphens, ie. `--tm-host`. Python's argparse will convert hypens (`-`) to underscores (`_`) for the variables, so no need to worry about it.